enhancements/sig-architecture/14-addon-cluster-proxy/README.md

```diff
@@ -0,0 +1,362 @@
+
+
+##### Direct HTTP calls
+
+Basically all the client library supporting HTTP proxy are expected to use 
+the proxy addons w/o additional work. As a matter of fact, most of those 
+popular HTTP client libraries only supports unsecured HTTP proxy instead of 
+mTLS-secured HTTPS proxy, and neither golang SDK nor JDK has builtin support 
+for mTLS proxy. So at the cost of securing the proxy server further, we will 
+need to adapt HTTPS proxy by our own effort. The good news is that the
+apiserver-network-proxy provided packages to talk over mTLS proxy. And also 
+there are a few https client binaries that natively supports mTLS proxy e.g.
+`curl` in the latest versions as is shown below:
+
+```bash
+curl -v -p \
+     --proxy-key certs/frontend/private/proxy-client.key \
+     --proxy-cert certs/frontend/issued/proxy-client.crt \
+     --proxy-cacert certs/frontend/issued/ca.crt \
+     --proxy-cert-type PEM \
+     -x https://127.0.0.1:8090 \   # the proxy-server's address
+     http://localhost:8000/success # any target L4 address
+````
+
+
+##### SSH over HTTPS tunnels
+
+The original ssh command supports customizing tunnelling techniques by passing
+`-o "ProxyCommand=..."` option, e.g. ssh over an plain HTTP proxy will be:
+
+```shell
+ssh user@host \
+-o "ProxyCommand=nc --proxy-type http --proxy 127.0.0.1:8090 %h %p"
+```
+
+However, mutual TLS is not yet supported for configuring proxy options, so an
+alternative will be replacing `nc` w/ a simple binary named `konnectivity-nc`
+that supports both mTLS https or grpc proxy as an extension.
+
+An very simple revision of `konnectiviy-nc` will be:
+
+[https://gist.github.com/yue9944882/3266bb3a73fde5538e861ae174f6f177](https://gist.github.com/yue9944882/3266bb3a73fde5538e861ae174f6f177)
+
+And the ssh command will be:
+
+```shell
+ssh user@host \
+-o "ProxyCommand=konnectivity-nc \
+--proxy-ca-cert <filepath to ca cert> \
+--proxy-cert <filepath to x509 cert> \
+--proxy-key <filepath to private key> \
+%h %p"
+```
+
+#### Framework Integration
+
+Based on the extensibility provided by the addon-framework, the new proxy addon
+will be implementing the required interfaces in the following behavior:
+
+
+```go
+var _ AgentAddon = &ApiserverNetworkProxyAddon{}
+
+func (a *ApiserverNetworkProxyAddon) Manifests(cluster *clusterv1.ManagedCluster, config runtime.Object) ([]runtime.Object, error) {
+    // 1. a service-account with fundamental spoke cluster permissions.
+    // 2. a deployment installing the proxy-agents
+}
+
+func (a *ApiserverNetworkProxyAddon) GetAgentAddonOptions() AgentAddonOptions {
+	// using the `ProxyConfiguration` above as options
+}
+
+var _ AgentAddonWithRegistration = &ApiserverNetworkProxyAddon{}
+
+func (a *ApiserverNetworkProxyAddon) HubRBAC(cluster *clusterv1.ManagedCluster, group string) (*rbacv1.Role, *rbacv1.RoleBinding) {
+	// none
+}
+
+func (a *ApiserverNetworkProxyAddon) BootstrapKubeConfig(cluster *clusterv1.ManagedCluster) ([]byte, error) {
```

Comment:
It seems not the latest addon framework interface. We simplify this so a bootstrapKubeConfig is not really needed.